neurom/core/morphology.py

```diff
@@ -211,7 +227,22 @@ def __repr__(self):
              NeuriteType.undefined: 4}
 
 
-def iter_neurites(obj, mapfun=None, filt=None, neurite_order=NeuriteIter.FileOrder):
+def _homogeneous_subtrees(neurite):
+    """Returns a list of the root nodes of the sub-neurites.
+
+    A sub-neurite can be either the entire tree or a homogeneous downstream
+    sub-tree.
+    """
```

Comment:
do we want to check/error/differenciate/document what happens if there is a `dendrite - axon - dendrite - axon` situation?

Reply:
Added a warning if the neurite is not an AcD

Reply:
Also, I changed the way that identifies the subtrees so that it creates a subtree for each different type it encounters downstream, regardless if it has been added to the list. This allows to have a `dendrite-axon-dendrite-axon` situation, which was not possible before due to the set I was using.